restli-client/src/main/java/com/linkedin/restli/client/Request.java

```diff
@@ -299,13 +302,29 @@ public UriTemplate getUriTemplate()
    */
   protected Set<PathSpec> getFields()
   {
-    @SuppressWarnings("unchecked")
-    Set<PathSpec> fieldsSet = (Set<PathSpec>) _queryParams.get(RestConstants.FIELDS_PARAM);
-    if (fieldsSet == null)
-    {
+    Object fields = _queryParams.get(RestConstants.FIELDS_PARAM);
+    if (fields == null) {
       return Collections.emptySet();
     }
-    return fieldsSet;
+
+    if (fields instanceof Set)
+    {
+      @SuppressWarnings("unchecked")
+      Set<PathSpec> fieldsSet = (Set<PathSpec>) _queryParams.get(RestConstants.FIELDS_PARAM);
+      return fieldsSet;
+    }
+    else if (fields instanceof String)
+    {
+      try
+      {
+        MaskTree tree = URIMaskUtil.decodeMaskUriFormat((String) fields);
+        return tree.getOperations().keySet();
+      } catch (IllegalMaskException e) {
+        throw new IllegalArgumentException(e);
```

Comment:
Please include a message:
"Field param was a string and it did not represent a serialized mask tree"